stylegan_code_finder/lightning_modules/base_lightning.py

```diff
@@ -1,41 +1,42 @@
-import os
-import torch
-from torch import nn
 import pytorch_lightning as pl
-from training_builder.base_train_builder import BaseTrainBuilder
-from networks.trans_u_net.utils import DiceLoss
 from torch.optim.lr_scheduler import CosineAnnealingWarmRestarts
 from utils.clamped_cosine import ClampedCosineAnnealingLR
 from visualization.segmentation_plotter_lightning import SegmentationPlotter
+from networks import load_weights
+from typing import List
+from torch.optim import Optimizer
 
 
 class BaseSegmenter(pl.LightningModule):
-    def __init__(self, training_builder: BaseTrainBuilder, configs, segmentation_plotter: SegmentationPlotter, wandb_logger):
+    def __init__(self, configs: dict):
         super().__init__()
-        self.segmentation_network = training_builder.segmentation_network
-        self.optimizers = list(training_builder.get_optimizers().values())
         self.configs = configs
-        self.updater = training_builder.get_updater()
+        self.segmentation_network = None
+        self._initialize_segmentation_network()
+        if configs['fine_tune'] is not None:
+            load_weights(self.segmentation_network, configs['fine_tune'], key='segmentation_network')

```

Comment:
For legacy reasons it might not be a bad idea to have this form of weight loading. However, we should definitely also have an option to load the Lightning Checkpoints (see `load_from_checkpoint()`)